dspace-api/src/main/java/org/dspace/browse/BrowsableObject.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.browse;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dspace.core.Constants;
+
+/**
+ * This is the basic interface that a data model entity need to implement to support browsing/retrieval
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ *
+ * @param <PK>
+ *            the Class of the primary key
+ */
+public interface BrowsableObject<PK extends Serializable> {
```

Comment:
Based on discussion today, I feel there's a lot of confusion around this new interface.   To me, it's clear we need a new interface, as there are some objects that we want to index in Solr that are *not* DSpaceObjects. So, we need a way to generically represent what a BrowsableObject is.

However, the methods this interface requires are unclear at times...I think it might be clarified by better Javadocs, see below inline comments...

Reply:
@tdonohue all the unnecessary methods have been removed as agreed via slack